pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -311,3 +318,46 @@ func checkWhiteList(c *m.ReqContext, host string) bool {
 
 	return true
 }
+
+func addOAuthPassThruAuth(c *m.ReqContext, req *http.Request) {
+	cmd := &m.GetAuthInfoQuery{UserId: c.UserId}
+	if err := bus.Dispatch(cmd); err != nil {
+		logger.Error("Error feching oauth information for user", "error", err)
+		return
+	}
+
+	provider := cmd.Result.AuthModule
+	connect, ok := social.SocialMap[strings.TrimPrefix(provider, "oauth_")] // The socialMap keys don't have "oauth_" prefix, but everywhere else in the system does
+	if !ok {
+		logger.Error("Failed to find oauth provider with given name", "provider", provider)
+		return
+	}
+
+	// TokenSource handles refreshing the token if it has expired
+	token, err := connect.TokenSource(c.Req.Context(), &oauth2.Token{
+		AccessToken:  cmd.Result.OAuthAccessToken,
+		Expiry:       cmd.Result.OAuthExpiry,
+		RefreshToken: cmd.Result.OAuthRefreshToken,
+		TokenType:    cmd.Result.OAuthTokenType,
+	}).Token()
+	if err != nil {
+		logger.Error("Failed to retrieve access token from oauth provider", "provider", cmd.Result.AuthModule)
+		return
+	}
+
+	// If the tokens are not the same, update the entry in the DB
+	if token.AccessToken != cmd.Result.OAuthAccessToken {
+		cmd2 := &m.UpdateAuthInfoCommand{
```

Comment:
`cmd2` is not a great name :) 
I know we have some cases of that already in the codebase but we dont have to introduce more.